outputs.tf

```diff
@@ -21,4 +21,9 @@ output "policy" {
 output "instance_profile" {
   description = "Name of the ec2 profile (if enabled)"
   value       = join("", aws_iam_instance_profile.default.*.name)
-}
\ No newline at end of file
+}
```

Comment:
Looks like tflint is failing above. Could you also change

```diff
output "instance_profile" {
  description = "Name of the ec2 profile (if enabled)"
- value       = join("", aws_iam_instance_profile.default.*.name)
+ value       = join("", aws_iam_instance_profile.default[*].name)
```

Reply:
All the variables, except the new one, follow the `.*.` pattern. Before I change more things in this PR, I would like to understand whether this approach is preferred over https://github.com/cloudposse/terraform-aws-iam-role/pull/86

Reply:
Yea makes sense to me. Unfortunately because the file didn't have a newline at the end of the file, tflint detects it as a change to the above block too which threw the error. If it's not corrected then the terratest will fail.

The two approaches do not have to be mutually exclusive. You can have both the object based inputs and the override. In the short term, 86 is preferable since it doesn't break the map. In the long run, when contributors are ready to release (could be today, I cannot say) a new major version of this module, this PR could be included or be the reason for it. Many times the contributors want to bundle a number of breaking changes when releasing a major version all at once so they don't have to release multiple major versions rapidly. 

cc @Benbentwo